lib/authz/project_purge/project_purge.go

```diff
@@ -0,0 +1,20 @@
+package project_purge
+
+import (
+	"context"
+	"math"
+	"time"
+)
+
+type PurgeClient interface {
+	PurgeProject(context.Context, string) error
```

Comment:
So i think this is already true for teams service, but make sure this function is repeatable (it doesn't blow up if a project does not exist)